core/storage/statistics/gae_models_test.py

```diff
@@ -43,6 +46,125 @@ def test_create_and_get_event_models(self):
         self.assertEqual(event_model.is_feedback_useful, True)
 
 
+class StateCounterModelTests(test_utils.GenericTestBase):
+
+    def test_stat_model_gets_created(self):
+        model_instance = stat_models.StateCounterModel.get_or_create(
+            'exp_id1', 'state_name')
+        self.assertEqual(model_instance.id, 'exp_id1.state_name')
+
+
+class ExplorationAnnotationsModelTests(test_utils.GenericTestBase):
+
+    def test_create_and_get_models(self):
+        stat_models.ExplorationAnnotationsModel.create('exp_id1', '1', 5, 4, {})
+
+        model1 = stat_models.ExplorationAnnotationsModel.get('exp_id1:1')
+
+        self.assertEqual(model1.exploration_id, 'exp_id1')
+        self.assertEqual(model1.version, '1')
+        self.assertEqual(model1.num_starts, 5)
+        self.assertEqual(model1.num_completions, 4)
+        self.assertEqual(model1.state_hit_counts, {})
+
+    def test_get_versions(self):
+        stat_models.ExplorationAnnotationsModel.create('exp_id1', '1', 5, 4, {})
+        stat_models.ExplorationAnnotationsModel.create('exp_id1', '2', 5, 4, {})
+
+        versions = stat_models.ExplorationAnnotationsModel.get_versions(
+            'exp_id1')
+
+        self.assertEqual(sorted(versions), ['1', '2'])
+
+
+class StateAnswersModelTests(test_utils.GenericTestBase):
+
+    def test_insert_submitted_answers(self):
```

Comment:
I don't really understand what this test is intended to test, it's doing quite a lot of things and the name doesn't help. Please clarify the name based on the guidelines laid down in the wiki page. Please do this for all test functions that you write.

Reply:
Done!